juju/juju
# Refactors model info copy for model db.

We require the need to have a model's metadata copied from the controller database to that of the model database. This is so that services local to the model database can have some basic information about how they should be operating in the current context.

Previously to do this we were copying the args that are used to make a model and writing those same values down into the model database. The problem with this approach is that the args are necessarily what is always written to the model database. A fair amount of business logic runs over the model creation args and there exist the possibility for computed values to become the new value and also for changes in the creation args over time.

What this commit attempts to do is change the process so that the copy of the information for the model database comes from a Get call after the model has been created.

This way we are always copying exactly what is in the controller database as an authoritative source. To do this however I have had to introduce some ugly channel code into the model bootstrap logic. This code is hard to read and can be error prone.

This commit is an attempt to show the problem and a possible solution to seed discussion on how best to solve this.

## Checklist


## QA steps

1. Bootstrap a new controller and confirm everything succeeds
2. Add a new model and confirm no issues.

## Documentation changes

N/A

## Links


**Jira card:** JUJU-
